flow/model/cdc_stream.go

```diff
@@ -39,8 +43,19 @@ func (r *CDCStream[T]) GetLastCheckpoint() int64 {
 	return r.lastCheckpointID.Load()
 }
 
-func (r *CDCStream[T]) AddRecord(record Record[T]) {
-	r.records <- record
+func (r *CDCStream[T]) AddRecord(ctx context.Context, record Record[T]) {
+	logger := logger.LoggerFromCtx(ctx)
+	for {
+		select {
+		case r.records <- record:
+			return
+		case <-time.After(10 * time.Second):
+			logger.Warn("waiting on adding record to stream", slog.Any("record", record))
```

Comment:
```suggestion
```
Adding check on context cancel is fair as receiver may leave channel blocked, but no need for time.After

Reply:
want it for logging purposes mainly, to debug future issues

will change to timer